packages/p2p-media-loader-core/src/types.ts

```diff
@@ -113,26 +115,36 @@ export type DynamicCoreConfig = Partial<
 /** Represents the configuration for the Core functionality that is common to all streams. */
 export type CommonCoreConfig = {
   /**
-   * Time after which a cached segment expires, in seconds.
-   * If set to undefined, the cacheSegmentExpiration is disabled for VOD streams, and a default value (20 minutes) is used for live streams.
+   * Maximum number of segments to store in the cache.
+   * Has to be less then httpDownloadTimeWindow and p2pDownloadTimeWindow.
+   * If set to 0, the cache is unlimited.
    *
    * @default
    * ```typescript
-   * cachedSegmentExpiration: undefined
+   * cachedSegmentsCount: 0
    * ```
    */
-  cachedSegmentExpiration?: number;
+  cachedSegmentsCount: number;
+
   /**
-   * Maximum number of segments to store in the cache.
-   * Has to be less then httpDownloadTimeWindow and p2pDownloadTimeWindow.
-   * If set to 0, the cache is unlimited.
+   * Custom storage class for VOD segments.
    *
    * @default
    * ```typescript
-   * cachedSegmentsCount: 0
+   * vodSegmentsStorage: undefined
    * ```
    */
-  cachedSegmentsCount: number;
+  vodSegmentsStorage?: (isLive: boolean) => SegmentsStorage;
+
+  /**
+   * Custom storage class for live segments.
+   *
+   * @default
+   * ```typescript
+   * liveSegmentsStorage: undefined
+   * ```
+   */
+  liveSegmentsStorage?: (isLive: boolean) => SegmentsStorage;
```

Comment:
I think we can combine it into one function now: segmentsStorage?: (isLive: boolean) => SegmentsStorage;